timed/permissions.py

```diff
@@ -141,3 +168,150 @@ class IsNotBilledAndVerfied(BasePermission):
+            return False
+
+        employment = employment_models.Employment.objects.get_at(
+            user=request.user, date=date.today()
+        )
+        if employment:
+            return not employment.is_external
+        return False  # pragma: no cover
+
+
+class IsExternal(IsAuthenticated):
+    """Allows access only to external employees."""
+
+    def has_permission(self, request, view):
+        if not super().has_permission(request, view):  # pragma: no cover
+            return False
+
+        if request.method not in SAFE_METHODS:
+            try:
+                employment = employment_models.Employment.objects.get_at(
+                    user=request.user, date=date.today()
+                )
+            except employment_models.Employment.DoesNotExist:  # pragma: no cover
+                raise PermissionDenied("User has no employment")
+
+            if employment:
+                return employment.is_external
+        return True
+
+    def has_object_permission(self, request, view, obj):
+        if not super().has_object_permission(request, view, obj):  # pragma: no cover
+            return False
+
+        employment = employment_models.Employment.objects.get_at(
+            user=request.user, date=date.today()
+        )
+        if employment:
+            return employment.is_external
+        return False  # pragma: no cover
+
+
+class IsManager(IsAuthenticated):
+    """Allows access only to assignees with manager role."""
+
+    def has_permission(self, request, view):
+        if not super().has_permission(request, view):  # pragma: no cover
+            return False
+
+        if request.method not in SAFE_METHODS:
+            return (
+                request.user.customer_assignees.filter(is_manager=True).exists()
+                or request.user.project_assignees.filter(is_manager=True).exists()
+                or request.user.task_assignees.filter(is_manager=True).exists()
+                or False
+            )
+
+    def has_object_permission(self, request, view, obj):
+        if not super().has_object_permission(request, view, obj):  # pragma: no cover
+            return False
+
+        user = request.user
+
+        if isinstance(obj, projects_models.Task):
+            return (
+                projects_models.Task.objects.filter(pk=obj.pk)
+                .filter(
+                    Q(task_assignees__user=user, task_assignees__is_manager=True)
+                    | Q(
+                        project__project_assignees__user=user,
+                        project__project_assignees__is_manager=True,
+                    )
+                    | Q(
+                        project__customer__customer_assignees__user=user,
+                        project__customer__customer_assignees__is_manager=True,
+                    )
+                )
+                .exists()
+            )
+        else:  # pragma: no cover
+            return False
```

Comment:
Throw an exception, similar to what you're doing in `IsReviewer`

Reply:
done